### PR TITLE
Remove ASTPrinter.

### PR DESCRIPTION
**Pull request by ekpyron**
Fixes https://github.com/ethereum/solidity/issues/6025 and Fixes https://github.com/ethereum/solidity/issues/5253.

### CODE REVIEW

**Comment by ekpyron** [comment]
Command line tests are killing me today...

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `2c70d8b2f0f6430a31ec60c3c485aa1a11ef4e55`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/recovery_ast_constructor/output:6: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:9: "exportedSymbols" : 
 test/cmdlineTests/recovery_ast_constructor/output:11: "Error1" : 
 test/cmdlineTests/recovery_ast_constructor/output:17: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:20: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:22: "literals" : 
 test/cmdlineTests/recovery_ast_constructor/output:35: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:37: "baseContracts" : 
 test/cmdlineTests/recovery_ast_constructor/output:41: "contractDependencies" : 
 test/cmdlineTests/recovery_ast_constructor/output:48: "linearizedBaseContracts" : 
 test/cmdlineTests/recovery_ast_constructor/output:55: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:58: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:64: "modifiers" : 
 test/cmdlineTests/recovery_ast_constructor/output:75: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:78: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:80: "parameters" : 
 test/cmdlineTests/recovery_ast_constructor/output:91: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:93: "parameters" : 
 test/cmdlineTests/recovery_ast_constructor/output:104: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:106: "statements" : 
 test/cmdlineTests/recovery_ast_constructor/output:122: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:128: "modifiers" : 
 test/cmdlineTests/recovery_ast_constructor/output:139: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:142: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:144: "parameters" : 
 test/cmdlineTests/recovery_ast_constructor/output:155: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:158: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:170: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:173: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:193: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:196: "attributes" : 
 test/cmdlineTests/recovery_ast_constructor/output:200: "children" : 
 test/cmdlineTests/recovery_ast_constructor/output:203: "attributes" : 

```
Please check that your changes are working as intended.

File: scripts/check_style.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 REPO_ROOT="$(dirname "$0")"/..
 cd $REPO_ROOT
 
-WHITESPACE=$(git grep -n -I -E "^.*[[:space:]]+$" | grep -v "test/libsolidity/ASTJSON\|test/libsolidity/ASTRecoveryTests\|test/compilationTests/zeppelin/LICENSE")
+WHITESPACE=$(git grep -n -I -E "^.*[[:space:]]+$" | grep -v "test/libsolidity/ASTJSON\|test/libsolidity/ASTRecoveryTests\|test/compilationTests/zeppelin/LICENSE\|test/cmdlineTests/recovery_ast_constructor/output")
```
Comment (ekpyron) [inline scripts/check_style.sh:9]: Adding a new exception here isn't nice, but I added https://github.com/ethereum/solidity/issues/7373 to track really fixing this.

**Review by chriseth — APPROVED** [review]
Ah, this makes my try-catch PR smaller :)

**Pull request merged**